sympy/simplify/cse_main.py

```diff
@@ -491,10 +491,10 @@ def tree_cse(exprs, symbols, opt_subs=None, order='canonical', ignore=()):
 
     ## Find repeated sub-expressions
 
-    to_eliminate = set()
+    to_eliminate = OrderedSet()
 
-    seen_subexp = set()
-    excluded_symbols = set()
+    seen_subexp = OrderedSet()
+    excluded_symbols = OrderedSet()
```

Comment:
Hey, what's happening down at line 527 where we have `list(map(...))` not being assigned to anything?

Reply:
map is an iterator in python3 that's why there is a `list` call to make sure it's evaluated.